repobee_sanitizer/sanitizer.py

```diff
@@ -72,79 +79,71 @@ class SanitizeRepo(plug.Plugin, plug.cli.Command):
     )
 
     def command(self) -> Optional[plug.Result]:
-        repo_root = self.repo_root.absolute()
-
-        input_error = self._validate_input(repo_root)
-        if input_error:
-            return input_error
-
-        if self.no_commit:
-            LOGGER.info("Executing dry run")
-            file_relpaths = _sanitize_repo.discover_dirty_files(repo_root)
-            errors = _sanitize_repo.sanitize_files(repo_root, file_relpaths)
-        else:
-            LOGGER.info(f"Sanitizing repo and updating {self.target_branch}")
-            effective_target_branch = self._resolve_effective_target_branch(
-                repo_root
+        try:
+            self._validate_input()
+            result_message = (
+                self._sanitize_no_commit()
+                if self.no_commit
+                else self._sanitize_to_target_branch()
             )
-            try:
-                errors = _sanitize_repo.sanitize_to_target_branch(
-                    repo_root, effective_target_branch, self.commit_message
-                )
-            except _gitutils.EmptyCommitError:
-                return plug.Result(
-                    name="sanitize-repo",
-                    msg="No diff between target branch and sanitized output. "
-                    f"No changes will be made to branch: {self.target_branch}",
-                    status=plug.Status.WARNING,
-                )
+            status = plug.Status.SUCCESS
+        except SanitizeError as err:
+            result_message = err.msg
+            status = plug.Status.ERROR
```

Comment:
Why not just return directly in the try/except? That gives two very clear, very different exits (because they are very different, one is an error exit and the other one is a success exit). 
```suggestion
            return plug.Result(name="sanitize-repo", msg=result_message, status=plug.Status.SUCCESS)
        except SanitizeError as err:
            return plug.Result(name="sanitize-repo", msg=result_message, status=plug.Status.ERROR)
```

Reply:
I felt one return statement was clearner. But when you think of them as completely different i think youre right